cheddar/cheddar-metrics-intercom/src/main/java/com/clicktravel/cheddar/metrics/intercom/IntercomMetricCollector.java

```diff
@@ -72,6 +78,47 @@ public void updateUser(final MetricUser user) {
     }
 
     @Override
+    public void addCustomAttributesToUser(final String userId, final Map<String, Object> customAttributes) {
+        try {
+            final User intercomUser = User.find(userId);
+
+            final List<Class<?>> classTypes = Arrays.asList(Boolean.class, String.class, Integer.class, Double.class,
+                    Long.class, Float.class);
+            final Optional<Entry<String, Object>> unsupportedClassType = customAttributes.entrySet().stream()
+                    .filter(entry -> !classTypes.contains(entry.getValue().getClass())).findAny();
+
+            if (unsupportedClassType.isPresent()) {
+                logger.debug("Unsupported custom attribute class type: " + unsupportedClassType.get());
+            } else {
+                customAttributes.entrySet().stream().forEach(entry -> {
+                    if (entry.getValue().getClass().equals(Boolean.class)) {
+                        intercomUser.addCustomAttribute(CustomAttribute.newBooleanAttribute((String) entry.getValue(),
+                                (Boolean) entry.getValue()));
```

Comment:
Shouldn't the first call `entry.getValue()` be `entry.getKey()` instead? The other branches in this method have the same comment. Also, the cast to `String` may not be needed, as the entry is of type `Map.Entry<String, Object>`